### DISCUSSION TITLE
Crate a parser for our moclojer/spec

### DISCUSSION THREAD

**Discussion by matheusfrancisco** [moclojer/moclojer#198] [Ideas]
The idea is to create a parser for moclojer spec, then we can use it in the editor also we can use in the codebase to validate and throw nice error information when the user send something invalid

**Comment by avelino** [comment]
reference basis for specification

* https://apiblueprint.org/documentation/specification.html
* vscode plugin https://github.com/XVincentX/vscode-apielements

**Reply by J0sueTM** [reply]
json schema can be used in order to inforce pre-defined tokens: https://json-schema.org/
spec: https://json-schema.org/specification

this extension from redhat, for example, uses json schema to lint yaml files: https://github.com/redhat-developer/yaml-language-server
